api/settings_api.py

```diff
@@ -27,16 +32,26 @@ def do_post(self, **kwargs):
     user_pref = user_models.UserPref.get_signed_in_user_pref()
     if not user_pref:
       self.abort(403, msg='User must be signed in')
-    new_notify = self.get_bool_param('notify')
-    user_pref.notify_as_starrer = new_notify
+    raw_data = self.request.json
+    new_notify = raw_data.get('notify')
+
+    if not isinstance(new_notify, bool):
```

Comment:
What happens if we don't have this explicit input validation and we just rely on `from_dict()`?

Reply:
Similarly, doesn't `get_bool_param()` have this check intrinsically?

Reply:
without this validation, tests would fail as bad requests caused by invalid input would not be raised (I think OAS would abort instead of raising a BadRequest exception). `get_bool_param()` helps get `notify` directly, but not from the model we generated

Reply:
We could always update the tests.  But, that can be deferred to a future PR.